crypto/fipsmodule/ml_kem/ml_kem_ref/kem.c

```diff
@@ -120,22 +143,58 @@ static int16_t centered_to_positive_representative(int16_t in) {
   return constant_time_select_int(mask, in, in_fixed);
 }
 
-#define BYTE_DECODE_12_OUT_SIZE (256)
-#define BYTE_DECODE_12_IN_SIZE  (32 * 12)
-#define BYTE_DECODE_12_NUM_BITS (256 * 12)
+#define BYTE_ENCODE_12_IN_SIZE  (256)
+#define BYTE_ENCODE_12_OUT_SIZE (32 * 12)
+#define BYTE_DECODE_12_OUT_SIZE (BYTE_ENCODE_12_IN_SIZE)
+#define BYTE_DECODE_12_IN_SIZE  (BYTE_ENCODE_12_OUT_SIZE)
+
+// FIPS 203. Algorithm 5 ByteEncode_12
+// Encodes an array of 256 12-bit integers into a byte array.
+// Intuition for the implementation:
+//   in:  |xxxxxxxxyyyy| |yyyyzzzzzzzz| ...
```

Comment:
I'm a bit confused how this is the same as the commented-out implementation: I thought that each 16-bit array member contained a 12-bit digit and the top 4 are discarded. 
The inner loops of the above implementations which looped over the bits was iterating up to 12, then control moved to the next element, unless I'm misunderstanding.

Reply:
> each 16-bit array member contained a 12-bit digit and the top 4 are discarded.

that's correct, and that's what we are doing here as well

Reply:
I see it now, thanks.